### PR TITLE
Add new rules from @typescript-eslint

### PR DESCRIPTION
**Pull request by barbogast**

### CODE REVIEW

**Comment by barbogast** [comment]
Note: `@typescript-eslint/no-this-alias` is removed since it is included in the 'recommended' set.

File: rules/typescript.js [hunk 1]
```diff
@@ -2,23 +2,34 @@ module.exports = {
   plugins: ['@typescript-eslint'],
   extends: ['plugin:@typescript-eslint/recommended'],
   rules: {
-    "@typescript-eslint/ban-ts-ignore": "off",
+    '@typescript-eslint/ban-ts-ignore': 'off',
     '@typescript-eslint/array-type': 'error',
     '@typescript-eslint/explicit-function-return-type': 'off',
-    '@typescript-eslint/member-delimiter-style': "off",
+    '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
```
Comment (yamov) [inline rules/typescript.js]: There may be a couple valid cases for when we may need interfaces (types and interfaces are not fully equivalent).
But I guess it is fine for now. We may not even even run into such situations any time soon. And if we do we can discuss this rule later.

File: rules/typescript.js [hunk 2]
```diff
@@ -2,23 +2,34 @@ module.exports = {
...
+    '@typescript-eslint/no-non-null-asserted-optional-chain': 'error',
     '@typescript-eslint/no-parameter-properties': 'error',
-    '@typescript-eslint/no-this-alias': 'error',
     '@typescript-eslint/no-unused-expressions': 'error',
     '@typescript-eslint/no-unused-vars': 'error',
     '@typescript-eslint/no-useless-constructor': 'error',
     '@typescript-eslint/prefer-for-of': 'error',
     '@typescript-eslint/prefer-optional-chain': 'error',
+
+    // rules which are not yet released, enable with next update
```
Comment (yamov) [inline rules/typescript.js]: Not released by @typescript-eslint you mean?

Comment (barbogast) [reply rules/typescript.js]: yes

**Pull request merged**